### PR TITLE
Adding additional patterns for decimal type

### PR DESCRIPTION
**Pull request by MinuMichaelOlassayil**
This is to support more patterns of decimal input from metadata.

### CODE REVIEW

File: grimlock-core/src/main/scala/commbank/grimlock/framework/Codec.scala [hunk 1]
```diff
@@ -166,7 +166,9 @@ object Codec {
       case BooleanCodec.Pattern() => BooleanCodec.fromShortString(str).map(Coproduct(_))
       case BoundedStringCodec.Pattern(_, _) => BoundedStringCodec.fromShortString(str).map(Coproduct(_))
       case DateCodec.Pattern(_) => DateCodec.fromShortString(str).map(Coproduct(_))
-      case DecimalCodec.Pattern(_, _) => DecimalCodec.fromShortString(str).map(Coproduct(_))
+      case DecimalCodec.patternGeneric => DecimalCodec.fromShortString(str).map(Coproduct(_))
```
Comment (poojke) [inline grimlock-core/src/main/scala/commbank/grimlock/framework/Codec.scala]: As below, ordering of patterns is important

File: version.sbt [hunk 2]
```diff
@@ -12,5 +12,5 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-version in ThisBuild := "0.7.4"
+version in ThisBuild := "0.7.5"
```
Comment (poojke) [inline version.sbt]: Please also update version in README.md

File: grimlock-core/src/test/scala/commbank/grimlock/TestCodec.scala [hunk 3]
```diff
@@ -98,6 +98,8 @@ class TestStringCodec extends TestGrimlock {
 
 class TestDecimalCodec extends TestGrimlock {
   val codec = DecimalCodec(5, 4)
+  val plainCodec = DecimalCodec.fromShortString("decimal")
+  val codecWithoutScale = DecimalCodec.fromShortString("decimal(10)")
```
Comment (poojke) [inline grimlock-core/src/test/scala/commbank/grimlock/TestCodec.scala]: No need to create val for something that's used only once. In test case below just do
```
DecimalCodec.fromShortString("decimal(10)") shouldBe Some(DecimalCodec(10, 0))
```

Comment (MinuMichaelOlassayil) [reply grimlock-core/src/test/scala/commbank/grimlock/TestCodec.scala]: Changed

File: version.sbt (hunk 2 above)
Comment (MinuMichaelOlassayil) [reply version.sbt]: Done

File: grimlock-core/src/main/scala/commbank/grimlock/framework/Codec.scala (hunk 1 above)
Comment (MinuMichaelOlassayil) [reply grimlock-core/src/main/scala/commbank/grimlock/framework/Codec.scala]: Changed the order

File: grimlock-core/src/main/scala/commbank/grimlock/framework/Codec.scala [hunk 4]
```diff
@@ -399,7 +403,9 @@ object DecimalCodec {
    * @return A `Some[DecimalCodec]` in case of success, `None` otherwise.
    */
   def fromShortString(str: String): Option[DecimalCodec] = str match {
-    case Pattern(precision, scale) =>
+    case `patternGeneric` => {for { p <- IntCodec.decode("38") } yield DecimalCodec(p, 0)}
```
[the remark this answers is not in the document]
Comment (MinuMichaelOlassayil) [reply grimlock-core/src/main/scala/commbank/grimlock/framework/Codec.scala]: order changed

File: grimlock-core/src/main/scala/commbank/grimlock/framework/Codec.scala [hunk 5]
```diff
@@ -389,7 +391,9 @@ case class DecimalCodec(precision: Int, scale: Int) extends Codec[BigDecimal] {
 /** Companion object to DecimalCodec. */
 object DecimalCodec {
   /** Pattern for parsing `DecimalCodec` from string. */
-  val Pattern = "decimal\\((\\d+),(\\d+)\\)".r
+  val patternWithScale = "decimal\\((\\d+)\\,\\s*(\\d+)\\)".r
```
[the remark this answers is not in the document]
Comment (MinuMichaelOlassayil) [reply grimlock-core/src/main/scala/commbank/grimlock/framework/Codec.scala]: done

**Pull request merged**